matrix/basic_matrix.py

```diff
@@ -29,7 +36,7 @@ def zero_matrix(self, row_dim, col_dim):
         """
         assert isinstance(row_dim, int) and isinstance(col_dim, int)
         ls_entries = [[0] * col_dim for _ in range(row_dim)]
-        return self.__class__(ls_entries=ls_entries)
+        return Matrix(ls_entries=ls_entries)
```

Comment:
I was hitting this error in the test_identity matrix before I changed this:

![image](https://user-images.githubusercontent.com/39224603/95372502-df56f000-0898-11eb-9ef5-88eafc55fb78.png)

Reply:
Hmm I probably set this up incorrectly originally. The reason I made it use __class__ instead of Matrix was because if we were in the symmetric matrix class I wanted the identity matrix to know that it is still symmetric and not just any possible matrix. That being said, I was just playing with random inheritance structures with this set-up so not even sure if this is doable to maintain.

Reply:
Also I might've also not updated the `test_identity` haha

Reply:
Ohhh, I see, that makes sense.  I changed back to __class__ for now, it just needed to be a static method